### PR TITLE
Add descriptions to dataset properties, store dataset schemas

### PR DESCRIPTION
**Pull request by spolu**
This PR adds the capability to add descriptions to Dust apps dataset properties. This is used to produce a DatasetSchema that is stored in DB and that we will be able to leverage to generate inputs to call the app from an custom Assistant.

The code of DatasetView is fucked beyond repair. After 3 attempts at refactoring it I gave up and added the feature in a minimally invasive way, keeping all the bad patterns that exist there for now :/

Deploy procedure:
- init_db on front
- deploy front

### CODE REVIEW

**Comment by spolu** [comment]
Screenshot from 2023-10-09 16-10-14

File: front/components/app/DatasetView.tsx [hunk 1]
```diff
@@ -351,18 +400,30 @@ export default function DatasetView({
...
-          MODELS_STRING_MAX_LENGTH
-        ),
-        data: exportDataset(),
-      });
+      onUpdate(
+        datasetInitializing,
+        valid,
+        {
+          name: datasetName.slice(0, MODELS_STRING_MAX_LENGTH),
+          // keys: datasetKeys,
```
Comment (fontanierh) [inline front/components/app/DatasetView.tsx:408]: Not sure I understand why this is commented out ?

File: front/lib/datasets.ts [hunk 2]
```diff
@@ -32,22 +38,33 @@ export function checkDatasetData(data?: Array<object> | null): string[] {
...
   datasetKeys: string[],
   entry: DatasetEntry
-): string[] {
+): ("string" | "number" | "boolean" | "json")[] {
   return datasetKeys.map((key) => getValueType(entry[key]));
 }
 
-export function getValueType(value: any): string {
+export function getValueType(
+  value: any
```
Comment (fontanierh) [inline front/lib/datasets.ts:62]: should we use `unknown` instead ?

File: front/pages/w/[wId]/a/[aId]/datasets/[name]/index.tsx [hunk 3]
```diff
@@ -140,14 +153,19 @@ export default function ViewDatasetView({
...
+          schema: updatedSchema,
+        }),
       }
     );
     await res.json();
     setEditorDirty(false);
     setIsFinishedEditing(true);
   };
 
+  console.log(schema);
```
Comment (fontanierh) [inline front/pages/w/[wId]/a/[aId]/datasets/[name]/index.tsx:167]: nit

**Review by fontanierh — DISMISSED** [review]

File: front/components/app/DatasetView.tsx (hunk 1 above)
Comment (spolu) [reply front/components/app/DatasetView.tsx:408]: We don't pass keys there, only the name description and data. This was previously commented. Will remove

File: front/lib/datasets.ts (hunk 2 above)
Comment (spolu) [reply front/lib/datasets.ts:62]: once we refactor that yes :+1:

**Pull request merged**